libsolidity/codegen/ArrayUtils.cpp

```diff
@@ -857,13 +860,17 @@ void ArrayUtils::popStorageArrayElement(ArrayType const& _type) const
 
 	if (_type.isByteArray())
 	{
-		m_context.appendInlineAssembly(R"({
+		util::Whiskers code(R"({
 			let slot_value := sload(ref)
 			switch and(slot_value, 1)
 			case 0 {
 				// short byte array
 				let length := and(div(slot_value, 2), 0x1f)
-				if iszero(length) { invalid() }
+				if iszero(length) {
+					mstore(0, <panicSelector>)
+					mstore(4, <emptyArrayPop>)
+					revert(0, 0x24)
```

Comment:
Would it make sense to extract these 3 lines similar to `CompilerContext::appendPanic`?

Reply:
Ah `YulUtilFunctions::panicFunction` has that

Reply:
How do you suggest to do it? I don't see a nice way to extract this without causing a function call.